src/middleware/validateScheme.js

```diff
@@ -0,0 +1,133 @@
+import { schemeFieldTypesEnum } from '../schemas/shared/schemeFieldTypesEnum.js';
+import { allRecordsExist, recordExist } from '../helpers/index.js';
+
+export function validateScheme(scheme) {
+  return async (req, res, next) => {
+    await checkObjectConfigs(res, req.body, scheme.body);
+    await checkObjectConfigs(res, req.query, scheme.query);
+
+    next();
+  };
+}
+
+async function checkObjectConfigs(res, objectToCheck, schemeConfigs) {
+  for (const pair of Object.entries(schemeConfigs)) {
+    const fieldName = pair[0];
+    const config = pair[1];
+
+    for (const configCheck of RequestFieldConfigsCheckers) {
+      if (configCheck[Symbol.toStringTag] === 'AsyncFunction') {
+        if (!(await configCheck(res, fieldName, objectToCheck, config))) return;
+      }
+
+      if (!configCheck(res, fieldName, objectToCheck, config)) return;
+    }
+  }
+}
+
+const RequestFieldConfigsCheckers = [
```

Comment:
Please, do not use PascalCase for JS variables - use camelCase or UPPER_CASE_SNAKE_CASE

Reply:
If it would be a class in TypeScript - then it would be absolutely correct to use PascalCase